docs/Manifests.md

```diff
@@ -1,6 +1,190 @@
 # Manifests
 
-## AppManifest
+## AppManifest Specification
+
+The AppManifest YAML file defines your application's components, routes, resources, and lifecycle events.
+
+### Components top-level element
+
+An AppManifest must declare a `components` root element as a list of objects containing configuration for each component in your application. At least one component object must be defined.
+
+```yaml
+components:
+  - name: webserver
+    ...
+  - name: api
+    ...
+```
+
+<!-- <details> -->
+
+#### name
+
+Defines the name of the component. Each component `name` must be unique. **Required**.
+
+#### type
+
+Specifies the type of the component; must be one of `service`, `task`, or `static`. **Required**.
+
+#### image
+
+Sets the base image from which the component is built. **Required**.
+
+#### port
+
+The container port to expose.
```

Comment:
Throughout this file we should provide default values (if they exist), along with what component types a specific directive can be used with.

Quick example here for `port` - The default value is `80`, and it is only a valid field for Service components. If someone's Service component is already using port `80` then there would be no need to explicitly define it within their AppManifest file.

... maybe this section should be broken up by Component type, or at least have a component specific sections providing further context on the differences between each component type 🤔

Reply:
Yes - thought about Component types after I submitted this. Think it should be broken up (or would have to be).  Assets would be required for statics, as an example, so in the sense it would have to be. Likewise for port, like you mentioned. So also understood here. 

Have to look back at this - I am not sure how I missed default values.